### PR TITLE
implement idempotent write for redis

### PR DESCRIPTION
**Pull request by wenxuanguan**
## What changes were proposed in this pull request?

implement idempotent write for redis

### CODE REVIEW

File: emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/package.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.apache.spark.sql.aliyun
```
Comment (uncleGen) [inline emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/package.scala]: miss license header

File: emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/RedisSink.scala [hunk 2]
```diff
@@ -18,26 +18,53 @@ package org.apache.spark.sql.aliyun.redis
...
-    val saveMode = sqlContext.sparkSession.conf
-      .getOption("redis.save.mode")
-      .getOrElse("append").toLowerCase match {
-      case "append" => SaveMode.Append
-      case "overwrite" => SaveMode.Overwrite
-      case "errorifexists" => SaveMode.ErrorIfExists
-      case "ignore" => SaveMode.Ignore
-      case unknown: String => throw new Exception(s"Unknown redis save mode $unknown.")
-    }
+  private var initialed = false
```
Comment (uncleGen) [inline emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/RedisSink.scala]: rename `initialed` to `restart`?

File: emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/package.scala (hunk 1 above)
Comment (wenxuanguan) [reply emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/package.scala]: fixed

File: emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/RedisSink.scala (hunk 2 above)
Comment (wenxuanguan) [reply emr-redis/src/main/scala/org/apache/spark/sql/aliyun/redis/RedisSink.scala]: add annotation

**Pull request merged**